lib/AnalysisStructured/PtrAnalysis.cpp

```diff
@@ -953,167 +993,81 @@ LogicalResult PtrAnalysis::rewriteForOp(scf::ForOp op) {
     //   // more operations to update %offset
     // }
     // If %tensor_of_ptr is not used (i.e., %tensor_of_ptr is updated before
-    // used in the original IR), it will simply be removed by canonicalization.
+    // used in the original IR), it will simply be removed by
+    // canonicalization.
 
     // For scalar pointers, there is no need to create a tts.addptr at the
     // beginning of the loop body. We don't lower tt.load and tt.store on
     // scalars in this pass; pointer arithmetics can also just use the
     // original pointer.
     if (state.getRank() != 0) {
-      OpBuilder::InsertionGuard guard(builder);
-      builder.setInsertionPointToStart(&newOp.getRegion().front());
+      OpBuilder builder(op.getRegion());
       auto maketptrOp = state.createTTSMakeTensorPtrOp(builder, op.getLoc());
       ptrMap.map(key, maketptrOp.getResult());
     }
   }
 
-  for (auto &bodyOp : newOp.getRegion().getOps()) {
-    if (auto forOp = dyn_cast<scf::ForOp>(bodyOp)) {
-      forOp->emitRemark("PtrAnalysis: nested loops currently not supported");
-      return failure();
-    }
-  }
-
-  // Update the loop body.
-  if (rewriteOp(newOp).failed()) {
-    newOp->erase();
+  // Recursively rewrite the inner ops
+  if (rewriteOp(op).failed()) {
+    op->erase();
```

Comment:
iiuc, this deletes the original for op? Why do we want to do that?

Reply:
yeah this is my mistake